### ISSUE TITLE
Rapid collisions cause massive amounts of lag.

### ISSUE THREAD

**Issue by Esnardo** [MovingBlocks/DestinationSol#556]
### How to reproduce: This format doesn't seem very good for this issue, let me just

Issue: Repeated collisions are incredibly taxing on the cpu, causing drops to FPS values better measured in SPF

1. Follow around a shop ship until it gets to a planet
1. Wait for it to get stuck in some corner somewhere (should only take a few seconds, maybe fix that too)
5. Try to do anything and notice it takes orders of magnitude longer because of the lag

Alternatively:

1. Be a large ship
2. Go to a mine and find a small gap
7. Get stuck 
3. Watch the fireworks as your CPU explodes

**Comment by Esnardo** [comment]
Huh. I don't know how to add the bug label

**Comment by BenjaminAmos** [comment]
I can reproduce this. Whenever this happens, I get a flood of sound-related error messages in the logcat.
```logcat
2023-03-20 20:16:06.646 27642-27685 AudioTrack              com.miloshpetrov.sol2.android        W  AUDIO_OUTPUT_FLAG_FAST denied by client; transfer 4, track 44143 Hz, output 48000 Hz
2023-03-20 20:16:07.060 27642-27685 AudioTrack              com.miloshpetrov.sol2.android        W  AUDIO_OUTPUT_FLAG_FAST denied by client; transfer 4, track 44911 Hz, output 48000 Hz
2023-03-20 20:16:07.073 27642-27685 AudioTrack              com.miloshpetrov.sol2.android        W  AUDIO_OUTPUT_FLAG_FAST denied by client; transfer 4, track 43552 Hz, output 48000 Hz
2023-03-20 20:16:07.074 27642-27685 AudioTrack              com.miloshpetrov.sol2.android        E  AudioFlinger could not create track, status: -12
2023-03-20 20:16:07.074 27642-27685 SoundPool               com.miloshpetrov.sol2.android        E  Error creating AudioTrack
```

Commenting-out the code the plays collision sounds appears to fix the issue, although I've only tested this lightly so-far.
https://github.com/MovingBlocks/DestinationSol/blob/fa2a67e059b8bbd6d7a25836d04d999ffb0b860b/engine/src/main/java/org/destinationsol/game/SolContactListener.java#L128-L129

A proper fix for this would probably involve deferring collision sound playback to outside of the collision handler. Re-sampling all the sound effects to 48000Hz might possibly help too but is a bit of a drastic measure.

**Comment by Esnardo** [comment]
Perhaps you could change the sound handler or something, make it rate limit
the number of sounds that can play at once?

**Comment by BenjaminAmos** [comment]
> Perhaps you could change the sound handler or something, make it rate limit the number of sounds that can play at once?

That's a good idea and would probably have helped but I discovered something else whilst looking into this.

It turns-out that playing sound effects blocks the current thread. It looks like this is actually a symptom of a very well known issue with the libGDX framework that we're using. The suggested fix is to [use the asynchronous audio back-end instead](https://libgdx.com/wiki/audio/audio#audio-on-android), which does appear to fix this issue for me. I've added the fix.
